api/cpp/include/stm32/slint-stm32.h

```diff
@@ -0,0 +1,176 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+#pragma once
+
+#define SLINT_STRINGIFY(X) SLINT_STRINGIFY2(X)
+#define SLINT_STRINGIFY2(X) #X
+#define SLINT_CAT(X, Y) SLINT_CAT2(X, Y)
+#define SLINT_CAT2(X, Y) X##Y
+
+#include <slint-platform.h>
+#include <cstdint>
+#include <memory>
+#include <type_traits>
```

Comment:
move these includes before the #define ?   so that reduces the scope of #define/#undef